tests/integration-tests/clusters_factory.py

```diff
@@ -285,12 +301,41 @@ def cfn_resources(self):
             self.__cfn_resources = retrieve_cfn_resources(self.cfn_name, self.region)
         return self.__cfn_resources
 
+    @property
+    def cfn_stack_arn(self):
+        """Return the CloudFormation stack ARN for the cluster."""
+        if not self.__cfn_stack_arn:
+            self.__cfn_stack_arn = (
+                boto3.client("cloudformation", region_name=self.region)
+                .describe_stacks(StackName=self.name)
+                .get("Stacks")[0]["StackId"]
+            )
+        return self.__cfn_stack_arn
+
     def _reset_cached_properties(self):
         """Discard cached data."""
         self.__cfn_parameters = None
         self.__cfn_outputs = None
         self.__cfn_resources = None
 
+    def _delete_resource_by_stack_id_tag(self):
+        self._delete_snapshots()
+        self._delete_volumes()
+
+    @suppress_and_log_exception
+    def _delete_snapshots(self):
+        ec2_client = boto3.client("ec2")
```

Comment:
do we need to set the region? same in the other function

Reply:
IMO, it is optional. But I added them

Reply:
I believe without region it breaks if running tests in sequential mode when multiple regions tests are executed in the same process, or are we properly exporting the AWS_DEFAULT_REGION var before each test?